ServerCore/Pages/Hints/CreateBulk.cshtml.cs

```diff
@@ -0,0 +1,138 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using ServerCore.DataModel;
+using ServerCore.ModelBases;
+
+namespace ServerCore.Pages.Hints
+{
+    [Authorize(Policy = "IsEventAdminOrAuthorOfPuzzle")]
+    public class CreateBulkModel : EventSpecificPageModel
+    {
+        public CreateBulkModel(PuzzleServerContext serverContext, UserManager<IdentityUser> userManager) : base(serverContext, userManager)
+        {
+        }
+
+        [BindProperty]
+        public bool DeleteExisting { get; set; }
+
+        [BindProperty]
+        public string Description { get; set; }
+
+        [BindProperty]
+        public string HintContent { get; set; }
+
+        [BindProperty]
+        public string Cost { get; set; }
+
+        [BindProperty]
+        public string DisplayOrder { get; set; }
+
+        public int PuzzleId { get; set; }
+
+        public Puzzle Puzzle { get; set; }
+
+        public async Task<IActionResult> OnGetAsync(int puzzleId)
+        {
+            PuzzleId = puzzleId;
+            Puzzle = await _context.Puzzles.Where(m => m.ID == puzzleId).FirstOrDefaultAsync();
+            return Page();
+        }
+
+        public async Task<IActionResult> OnPostAsync(int puzzleId)
+        {
+            if (!ModelState.IsValid)
+            {
+                return Page();
+            }
+
+            if (DeleteExisting)
+            {
+                Hint[] hintsToRemove = await (from Hint h in _context.Hints
+                                                      where h.PuzzleID == puzzleId
+                                                      select h).ToArrayAsync();
+                _context.Hints.RemoveRange(hintsToRemove);
+            }
+
+            StringReader descriptionReader = new StringReader(Description ?? string.Empty);
```

Comment:
StringReader is disposable. I'm not sure if it needs it in this case, but just to be safe, please add using to it. I think with .NET 5, you can just write

```using StringReader = new StringReader(Description ?? string.Empty);```

Reply:
Neat, thanks!